common/achievement/get.ts

```diff
@@ -1,14 +1,37 @@
 import { prisma } from '../prisma';
-import { User_achievement, achievement_action_type_enum, achievement_type_enum } from '../../graphql/generated';
+import { important_information_recipients_enum, User_achievement, achievement_action_type_enum, achievement_type_enum } from '../../graphql/generated';
 import { Achievement, achievement_state } from '../../graphql/types/achievement';
 import { User } from '../user';
 import { ConditionDataAggregations } from './types';
 import { getAchievementState, getCurrentAchievementTemplateWithContext, transformPrismaJson } from './util';
 import { evaluateAchievement } from './evaluate';
-import { Prisma } from '@prisma/client';
 
-// TODO: getAchievementById -> passed user and achievementId to return a single achievement
-// TODO: resolver for nextSteps -> get active sequential achievements and important information
+const getAchievementById = async (user: User, achievementId: number): Promise<Achievement> => {
+    const userAchievement = await prisma.user_achievement.findUnique({
+        where: { id: achievementId },
+        include: { template: true },
+    });
+    const achievement = await assembleAchievementData([userAchievement], user);
+    return achievement;
+};
+
+// Next step achievements are sequential achievements that are currently active and not yet completed. They get displayed in the next step card section.
+const getNextStepAchievements = async (user: User): Promise<Achievement[]> => {
+    const userAchievements = await prisma.user_achievement.findMany({
+        where: { userId: user.userID, isSeen: false, template: { type: achievement_type_enum.SEQUENTIAL } },
```

Comment:
do we also have to check here for `achievedAt != null`?

Reply:
No I don't think so. If that was the case we could not get all steps necessary to know the currentStep value. Or the maxSteps value.